### PR TITLE
[d8-control-plane-manager] crashes when trying to update config

### PR DESCRIPTION
**Pull request by borg-z**
## Description

This PR adds logic to recreate the kubeconfig for components like "super-admin", "admin", "controller-manager", and "scheduler" if the current kubeconfig fails validation. 

## Why do we need it, and what problem does it solve?

This change stops d8-control-plane-manager from crashing when the kubeconfig is invalid. By adding checks and automatically regenerating the kubeconfig, it ensures the control plane components keep working even if their configuration gets messed up or outdated.

## What is the expected result?

After applying these changes, if a kubeconfig is invalid, it will be automatically regenerated.

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: control-plane-manager
type: fix
summary: Automatically regenerate kubeconfig for control plane components if validation fails, preventing crashes.
impact: 
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/openstack'...

**Comment by RomanenkoDenys** [comment]
is config recreate in the case of fail is proper way ? Can we manually check and understand what's wrong ?

**Comment by borg-z** [comment]
> is config recreate in the case of fail is proper way ? Can we manually check and understand what's wrong ?

Yes, we have case when kubeconfig is broken and d8-control-plane-manager can't start and hangs in crash-loopback. So to avoid this problem automatically I suggested to recreate the config in case it is not valid

File: modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go [hunk 1]
```diff
@@ -65,73 +65,77 @@ func renewKubeconfig(componentName string) error {
...
+	// regenerate kubeconfig
+	log.Infof("generate new kubeconfig %s", path)
+	return prepareKubeconfig(componentName, false)
+}
 
-		if currentKubeconfig.Clusters[0].Cluster.Server != tmpKubeconfig.Clusters[0].Cluster.Server {
-			log.Infof("kubeconfig %s address field changed", path)
-			remove = true
-		}
+func validateKubeconfig(path string, tmpPath string) (error, bool) {
```
Comment (RomanenkoDenys) [inline modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go:92]: it's wrong way - you should return error and check error type. You do not need bool parameter here.

Comment (borg-z) [reply modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go]: https://github.com/deckhouse/deckhouse/pull/9445/commits/db5f1c108673fb1e77e5159de99597c41ba28362
please review

**Review by name212 — APPROVED** [review]

File: modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go [hunk 2]
```diff
@@ -28,13 +28,45 @@ import (
...
+	ErrUsersFieldEmpty           = errors.New("users field is empty")
+	ErrCertDataFieldEmpty        = errors.New("client-certificate-data field is empty")
+	ErrCertDecodeFailed          = errors.New("cannot PEM decode client-certificate-data")
+	ErrCertParseFailed           = errors.New("cannot parse certificate from client-certificate-data")
+	ErrServerAddressChanged      = errors.New("kubeconfig address field changed")
+	ErrCertExpiringSoon          = errors.New("client certificate is expiring in less than 30 days")
+	ErrCantReadOrUnmarshalConfig = errors.New("cannot read or unmarshal kubeconfig")
+)
+
+var shouldRecreteKubeConfigErrors = []error{
```
Comment (RomanenkoDenys) [inline modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go:50]:
```suggestion
var shouldRecreateKubeConfigErrors = []error{
```

File: modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go [hunk 3]
```diff
@@ -28,13 +28,45 @@ import (
...
+	ErrUsersFieldEmpty,
+	ErrCertDataFieldEmpty,
+	ErrCertDecodeFailed,
+	ErrCertParseFailed,
+	ErrServerAddressChanged,
+	ErrCertExpiringSoon,
+	ErrCantReadOrUnmarshalConfig,
+}
+
+func shouldRecreteKubeConfig(err error) bool {
```
Comment (RomanenkoDenys) [inline modules/040-control-plane-manager/images/control-plane-manager/controller/kubeconfig.go:61]:
```suggestion
func shouldRecreateKubeConfig(err error) bool {
```

**Review by RomanenkoDenys — APPROVED** [review]
Please fix spelling errors

**Pull request merged**